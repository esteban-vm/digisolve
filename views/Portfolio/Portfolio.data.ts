import type { WorkProps } from './Work'

const width = 394
const height = 470

const works: WorkProps[] = [
  {
    id: '0a6dd16e-0229-4898-a3b4-b6b8a2dd5e5a',
    text: 'Online advertising',
    image: { src: '/img/online-advertising.png', width, height },
  },
  {
    id: '35501350-65b2-47bc-95a8-97bc27d2e186',
    text: 'Social media campaigns',
    image: { src: '/img/social-media-campaigns.png', width, height },
  },
  {
    id: '26c86632-8302-4ba4-b316-0a810712efef',
    text: 'UX design',
    image: { src: '/img/ux-design.png', width, height },
  },
  {
    id: 'e333c170-6d2a-43e9-9720-0a1d50d9c3ba',
    text: 'Branding & digital marketing',
    image: { src: '/img/branding-digital-marketing.png', width, height },
  },
]

export default works
