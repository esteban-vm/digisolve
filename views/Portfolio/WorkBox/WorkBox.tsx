import type { FunctionalComponent } from '@/types'
import Image, { type StaticImageData } from 'next/image'
import { styled } from '@/styles'

/** Properties for work box component */
export type WorkProps = {
  /** The image */
  image: StaticImageData
  /** The text */
  text: string
}

const WorkBoxComponent: FunctionalComponent<WorkProps> = ({ image, text, ...rest }) => {
  return (
    <figure {...rest}>
      <Image src={image} alt={text} />
      <p>{text}</p>
    </figure>
  )
}

const WorkBox = styled(WorkBoxComponent)`
  background-color: var(--color-black);
  overflow: hidden;

  img {
    width: 100%;
    height: auto;
    opacity: 0.7;
    transform: scale(1.15);
    transition: transform 500ms, opacity 500ms;

    :hover {
      transform: scale(1);
      opacity: 1;
    }
  }

  p {
    color: var(--color-white);
  }
`

export default WorkBox
