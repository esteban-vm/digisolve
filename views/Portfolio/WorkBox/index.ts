export { default, type WorkProps } from './WorkBox'
