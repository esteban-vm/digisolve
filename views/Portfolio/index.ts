export { default as Portfolio, works } from './Portfolio'
