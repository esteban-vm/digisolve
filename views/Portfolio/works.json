[
  {
    "title": "Online advertising",
    "image": {
      "src": "/img/online-advertising.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "Social media campaigns",
    "image": {
      "src": "/img/social-media-campaigns.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "UX design",
    "image": {
      "src": "/img/ux-design.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "Branding & digital marketing",
    "image": {
      "src": "/img/branding-digital-marketing.png",
      "height": 470,
      "width": 394
    }
  }
]
