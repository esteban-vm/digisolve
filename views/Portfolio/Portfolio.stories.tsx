import type { Meta } from '@storybook/react'
import Portfolio from './Portfolio'

export default {
  component: Portfolio,
  title: 'Views/Portfolio',
} as Meta<typeof Portfolio>

export const PortfolioView = () => <Portfolio />
