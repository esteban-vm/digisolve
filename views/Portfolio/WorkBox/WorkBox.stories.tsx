import type { Meta, StoryObj } from '@storybook/react'
import { within, userEvent } from '@storybook/testing-library'
import { styled } from '@/utils/styles'
import { works } from '../Portfolio'
import WorkBox from './WorkBox'

const Wrapper = styled.div`
  text-align: center;
`

export default {
  argTypes: {
    image: {
      control: false,
    },
    onClick: {
      action: true,
      control: false,
    },
    onMouseEnter: {
      action: true,
      control: false,
    },
  },
  component: WorkBox,
  title: 'Views/Portfolio/Works',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Story {...context} />
        </Wrapper>
      )
    },
  ],
  parameters: {
    layout: 'centered',
  },
  play({ canvasElement }) {
    const canvas = within(canvasElement)
    const image = canvas.getByRole('img')
    userEvent.click(image)
    userEvent.hover(image)
  },
} as Meta<typeof WorkBox>

const [work1, work2, work3, work4] = works

export const WorkExample1: StoryObj<typeof WorkBox> = { args: work1 }
export const WorkExample2: StoryObj<typeof WorkBox> = { args: work2 }
export const WorkExample3: StoryObj<typeof WorkBox> = { args: work3 }
export const WorkExample4: StoryObj<typeof WorkBox> = { args: work4 }
