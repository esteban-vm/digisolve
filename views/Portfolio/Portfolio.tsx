import type { FunctionalComponent } from '@/types'
import { Heading } from '@/components'
import { advertising, campaigns, design, marketing } from '@/utils/images'
import { styled } from '@/utils/styles'
import WorkBox, { type WorkProps } from './WorkBox'

export const works: WorkProps[] = [
  { image: advertising, text: 'Online Advertising' },
  { image: campaigns, text: 'Social Media Campaigns' },
  { image: design, text: 'UX Design' },
  { image: marketing, text: 'Branding & Digital Marketing' },
]

const PortfolioComponent: FunctionalComponent = (props) => {
  return (
    <section {...props}>
      <Heading heading='Our Work' />
      <ul className='clear_fix'>
        {works.map((work, index) => (
          <li key={index}>
            <WorkBox {...work} />
          </li>
        ))}
      </ul>
    </section>
  )
}

const Portfolio = styled(PortfolioComponent)`
  text-align: center;
  background: var(--color-light);

  ul {
    margin-top: 10px;
    list-style: none;
    width: 100%;

    li {
      display: block;
      width: 25%;
      float: left;
    }
  }
`

export default Portfolio
