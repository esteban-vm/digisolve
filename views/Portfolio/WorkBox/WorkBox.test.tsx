import type { ReactTestRendererJSON } from 'react-test-renderer'
import { render, cleanup, create, screen } from '@/utils/tests'
import { works } from '../Portfolio'
import WorkBox from './WorkBox'

describe('ðŸ§ª <WorkBox /> test cases:', () => {
  describe('there should be:', () => {
    beforeEach(() => {
      render(<WorkBox {...works[0]} />)
    })

    afterEach(cleanup)

    it('an image', () => {
      const image = screen.getByRole('img')
      expect(image).toBeInTheDocument()
      expect(image).toBeVisible()
      expect(image).toHaveAccessibleName(/^online\sadvertising$/i)
    })

    it('a text', () => {
      const text = screen.getByRole('paragraph')
      expect(text).toBeInTheDocument()
      expect(text).toBeVisible()
      expect(text).toHaveTextContent(/^online\sadvertising$/i)
    })
  })

  describe('style tests:', () => {
    let workTree: ReactTestRendererJSON

    beforeEach(() => {
      workTree = create(<WorkBox {...works[0]} />).toJSON() as ReactTestRendererJSON
    })

    it('should render with correct styles', () => {
      expect(workTree).toMatchSnapshot()
    })

    it('should transform scale and increment opacity when hovered', () => {
      expect(workTree).toHaveStyleRule('transform', 'scale(1)', { target: 'img:hover' })
      expect(workTree).toHaveStyleRule('opacity', '1', { target: 'img:hover' })
    })
  })
})
