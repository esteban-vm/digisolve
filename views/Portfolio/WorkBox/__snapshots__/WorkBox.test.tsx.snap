// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª <WorkBox /> test cases: style tests: should render with correct styles 1`] = `
.emotion-0 {
  background-color: var(--color-black);
  overflow: hidden;
}

.emotion-0 img {
  width: 100%;
  height: auto;
  opacity: 0.7;
  -webkit-transform: scale(1.15);
  -moz-transform: scale(1.15);
  -ms-transform: scale(1.15);
  transform: scale(1.15);
  -webkit-transition: -webkit-transform 500ms,opacity 500ms;
  transition: transform 500ms,opacity 500ms;
}

.emotion-0 img:hover {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 1;
}

.emotion-0 p {
  color: var(--color-white);
}

<figure
  className="emotion-0"
>
  <img
    alt="Online Advertising"
    data-nimg="1"
    decoding="async"
    height={40}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={40}
  />
  <p
    role="paragraph"
  >
    Online Advertising
  </p>
</figure>
`;
